### PR TITLE
Add support for multiple OpenID providers with role policies

### PR DESCRIPTION
**Pull request by donatello**
## Description

- When using multiple providers, claim based providers are not allowed. All
providers must use role policies.

Console changes are not done in this change - so it is not yet possible to login via the web console.

## Motivation and Context

Extend role policy support to enable multiple OpenID provider/apps.

## How to test this PR?

See examples in the doc update in the PR. 

Test setup can be done with dex using the containers and make file in https://github.com/minio/minio-iam-testing - see readme: https://github.com/minio/minio-iam-testing/tree/main/openid#running-minio-with-multiple-oidc-providers

STS credentials can be generated programmatically with the program at `/docs/site-replication/gen-oidc-sts-cred.go` from this PR.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [x] Documentation updated
- [x] Unit tests added/updated

### CODE REVIEW

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by donatello** [comment]
PTAL - ready for review

File: cmd/sts-handlers.go [hunk 1]
```diff
@@ -365,54 +367,11 @@ func (sts *stsAPIHandlers) AssumeRoleWithSSO(w http.ResponseWriter, r *http.Requ
...
-				errors.New("STS JWT Token has `aud` claim invalid, `aud` must match configured OpenID Client ID"))
-			return
-		}
-		if !azpValues.Contains(globalOpenIDConfig.ClientID) {
-			writeSTSErrorResponse(ctx, w, true, ErrSTSInvalidParameterValue,
-				errors.New("STS JWT Token has `azp` claim invalid, `azp` must match configured OpenID Client ID"))
-			return
-		}
-	}
-
```
Comment (harshavardhana) [inline cmd/sts-handlers.go:403]: Not sure why this is removed @donatello - not able to find a good explanation here.

File: cmd/sts-handlers_test.go [hunk 2]
```diff
@@ -989,9 +989,46 @@ var testAppParams = OpenIDClientAppParams{
 }
 
 const (
-	EnvTestOpenIDServer = "OPENID_TEST_SERVER"
+	EnvTestOpenIDServer  = "OPENID_TEST_SERVER"
+	EnvTestOpenIDServer2 = "OPENID_TEST_SERVER_2"
```
Comment (harshavardhana) [inline cmd/sts-handlers_test.go:1006]: You can perhaps now have `_1` and `_2` which would make sense.

File: internal/config/identity/openid/jwt.go [hunk 3]
```diff
@@ -346,29 +400,64 @@ func (r *Config) Validate(token, accessToken, dsecs string) (map[string]interfac
...
+		// string containing a StringOrURI value
+		azpValues, ok := iampolicy.GetValuesFromClaims(claims, azpClaim)
+		if !ok {
+			return nil, errors.New("STS JWT Token has `azp` claim invalid, `azp` must match configured OpenID Client ID")
+		}
+		if !azpValues.Contains(pCfg.ClientID) {
+			return nil, errors.New("STS JWT Token has `azp` claim invalid, `azp` must match configured OpenID Client ID")
+		}
+	}
+
```
Comment (harshavardhana) [inline internal/config/identity/openid/jwt.go:440]: Oh I see its moved here now

**Review by harshavardhana — CHANGES_REQUESTED** [review]
Let's talk to @Alevsk today about this change and relevant changes needed from the console side as well to facilitate and use this feature.

**Comment by harshavardhana** [comment]
what is waiting on this PR ? @donatello

**Comment by donatello** [comment]
This has one change remaining, will do soon

**Comment by minio-trusted** [comment]
```
Error running mint automation
+ mkdir -p 14223-9dd2263/gopath/src/github.com/minio
+ git -C 14223-9dd2263/gopath/src/github.com/minio clone --quiet https://github.com/minio/minio.git
+ git -C 14223-9dd2263/gopath/src/github.com/minio/minio remote add minio https://github.com/minio/minio.git
+ git -C 14223-9dd2263/gopath/src/github.com/minio/minio fetch --quiet minio pull/14223/head:pr14223
+ git -C 14223-9dd2263/gopath/src/github.com/minio/minio checkout --quiet pr14223
+ GO111MODULE=on
+ GOPATH=/home/mint-auto/mint-auto/14223-9dd2263/gopath
+ make -C 14223-9dd2263/gopath/src/github.com/minio/minio --quiet
go: downloading google.golang.org/api v0.67.0
go: downloading github.com/minio/sha256-simd v1.0.0
go: downloading github.com/minio/minio-go/v7 v7.0.23
go: downloading cloud.google.com/go/storage v1.10.0
go: downloading github.com/Azure/azure-pipeline-go v0.2.2
go: downloading github.com/dustin/go-humanize v1.0.0
go: downloading github.com/rs/dnscache v0.0.0-20211102005908-e0241e321417
go: downloading github.com/json-iterator/go v1.1.12
go: downloading github.com/minio/madmin-go v1.3.5
go: downloading github.com/secure-io/sio-go v0.3.1
go: downloading github.com/minio/sio v0.3.0
go: downloading github.com/minio/cli v1.22.0
go: downloading github.com/minio/kes v0.18.0
go: downloading golang.org/x/sys v0.0.0-20220128215802-99c3d69c2c27
go: downloading github.com/jcmturner/gokrb5/v8 v8.4.2
go: downloading github.com/golang-jwt/jwt/v4 v4.2.0
go: downloading github.com/colinmarc/hdfs/v2 v2.2.0
go: downloading golang.org/x/crypto v0.0.0-20220128200615-198e4374d7ed
go: downloading github.com/fatih/color v1.13.0
go: downloading github.com/buger/jsonparser v1.1.1
go: downloading github.com/Shopify/sarama v1.30.0
go: downloading github.com/klauspost/compress v1.14.4
go: downloading github.com/xdg/scram v0.0.0-20180814205039-7eeb5667e42c
go: downloading github.com/minio/highwayhash v1.0.2
go: downloading github.com/minio/pkg v1.1.20
go: downloading github.com/Azure/azure-storage-blob-go v0.10.0
go: downloading github.com/bits-and-blooms/bloom/v3 v3.0.1
go: downloading github.com/cespare/xxhash/v2 v2.1.2
go: downloading github.com/bcicen/jstream v1.0.1
go: downloading github.com/cheggaaa/pb v1.0.29
go: downloading github.com/dchest/siphash v1.2.1
go: downloading github.com/coreos/go-oidc v2.1.0+incompatible
go: downloading github.com/gorilla/mux v1.8.0
go: downloading github.com/cosnicolaou/pbzip2 v1.0.1
go: downloading github.com/hashicorp/golang-lru v0.5.4
go: downloading github.com/inconshreveable/mousetrap v1.0.0
go: downloading github.com/klauspost/reedsolomon v1.9.15
go: downloading github.com/google/uuid v1.3.0
go: downloading github.com/klauspost/readahead v1.4.0
go: downloading github.com/djherbis/atime v1.0.0
go: downloading github.com/felixge/fgprof v0.9.2
go: downloading github.com/minio/dperf v0.3.4
go: downloading github.com/minio/console v0.15.1
go: downloading github.com/miekg/dns v1.1.46
go: downloading github.com/go-openapi/loads v0.21.0
go: downloading github.com/klauspost/pgzip v1.2.5
go: downloading github.com/minio/selfupdate v0.4.0
go: downloading github.com/minio/zipindex v0.2.1
go: downloading github.com/mitchellh/go-homedir v1.1.0
go: downloading github.com/philhofer/fwd v1.1.2-0.20210722190033-5c56ac6d0bb9
go: downloading github.com/pierrec/lz4 v2.6.1+incompatible
go: downloading github.com/prometheus/client_golang v1.12.1
go: downloading github.com/prometheus/client_model v0.2.0
go: downloading github.com/prometheus/procfs v0.7.3
go: downloading github.com/rs/cors v1.7.0
go: downloading github.com/shirou/gopsutil/v3 v3.21.12
go: downloading github.com/tinylib/msgp v1.1.7-0.20211026165309-e818a1881b0e
go: downloading github.com/valyala/bytebufferpool v1.0.0
go: downloading github.com/yargevad/filepathx v1.0.0
go: downloading github.com/zeebo/xxh3 v1.0.0
go: downloading go.etcd.io/etcd/api/v3 v3.5.2
go: downloading go.etcd.io/etcd/client/v3 v3.5.2
go: downloading go.uber.org/atomic v1.9.0
go: downloading golang.org/x/oauth2 v0.0.0-20211104180415-d3ed0bb246c8
go: downloading golang.org/x/time v0.0.0-20211116232009-f0f3c7e86c11
go: downloading go.uber.org/zap v1.19.1
go: downloading github.com/eclipse/paho.mqtt.golang v1.3.5
go: downloading github.com/elastic/go-elasticsearch/v7 v7.12.0
go: downloading github.com/go-sql-driver/mysql v1.6.0
go: downloading github.com/gomodule/redigo v1.8.8
go: downloading github.com/lib/pq v1.10.4
go: downloading github.com/nats-io/nats.go v1.13.1-0.20220121202836-972a071d373d
go: downloading github.com/nats-io/stan.go v0.10.2
go: downloading github.com/nsqio/go-nsq v1.0.8
go: downloading github.com/pkg/errors v0.9.1
go: downloading github.com/streadway/amqp v1.0.0
go: downloading github.com/beevik/ntp v0.3.0
go: downloading github.com/go-ldap/ldap/v3 v3.2.4
go: downloading github.com/coredns/coredns v1.9.0
go: downloading github.com/golang-jwt/jwt v3.2.2+incompatible
go: downloading github.com/minio/simdjson-go v0.4.2
go: downloading github.com/montanaflynn/stats v0.6.6
go: downloading github.com/ncw/directio v1.0.5
go: downloading github.com/mattn/go-ieproxy v0.0.1
go: downloading golang.org/x/sync v0.0.0-20210220032951-036812b2e83c
go: downloading github.com/pquerna/cachecontrol v0.0.0-20171018203845-0dec1b30a021
go: downloading gopkg.in/square/go-jose.v2 v2.5.1
go: downloading github.com/golang/protobuf v1.5.2
go: downloading github.com/mattn/go-colorable v0.1.12
go: downloading github.com/mattn/go-isatty v0.0.14
go: downloading github.com/mattn/go-runewidth v0.0.13
go: downloading github.com/xdg/stringprep v1.0.0
go: downloading github.com/klauspost/cpuid/v2 v2.0.11
go: downloading github.com/minio/csvparser v1.0.0
go: downloading github.com/minio/parquet-go v1.1.0
go: downloading github.com/alecthomas/participle v0.2.1
go: downloading golang.org/x/net v0.0.0-20220127200216-cd36cc0744dd
go: downloading github.com/minio/argon2 v1.0.0
go: downloading github.com/golang/snappy v0.0.4
go: downloading github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: downloading github.com/modern-go/reflect2 v1.0.2
go: downloading github.com/gogo/protobuf v1.3.2
go: downloading github.com/nats-io/nuid v1.0.1
go: downloading github.com/bits-and-blooms/bitset v1.2.0
go: downloading github.com/go-openapi/analysis v0.21.2
go: downloading github.com/go-openapi/spec v0.20.4
go: downloading github.com/go-openapi/swag v0.19.15
go: downloading go.etcd.io/etcd/client/pkg/v3 v3.5.2
go: downloading google.golang.org/grpc v1.44.0
go: downloading github.com/google/pprof v0.0.0-20211214055906-6f57359322fd
go: downloading github.com/rivo/uniseg v0.2.0
go: downloading golang.org/x/text v0.3.7
go: downloading github.com/lestrrat-go/jwx v1.2.14
go: downloading github.com/fatih/structs v1.1.0
go: downloading gopkg.in/yaml.v2 v2.4.0
go: downloading github.com/rjeczalik/notify v0.9.2
go: downloading github.com/coreos/go-semver v0.3.0
go: downloading google.golang.org/genproto v0.0.0-20220207164111-0872dc986b00
go: downloading github.com/gorilla/websocket v1.4.2
go: downloading github.com/hashicorp/go-uuid v1.0.2
go: downloading github.com/jcmturner/gofork v1.0.0
go: downloading github.com/jcmturner/dnsutils/v2 v2.0.0
go: downloading github.com/Azure/go-ntlmssp v0.0.0-20200615164410-66371956d46c
go: downloading github.com/go-asn1-ber/asn1-ber v1.5.1
go: downloading github.com/prometheus/common v0.32.1
go: downloading github.com/beorn7/perks v1.0.1
go: downloading github.com/apache/thrift v0.15.0
go: downloading google.golang.org/protobuf v1.27.1
go: downloading github.com/mailru/easyjson v0.7.7
go: downloading github.com/nats-io/nkeys v0.3.0
go: downloading cloud.google.com/go v0.100.2
go: downloading github.com/rs/xid v1.3.0
go: downloading github.com/tidwall/gjson v1.12.1
go: downloading github.com/tidwall/sjson v1.2.3
go: downloading github.com/jcmturner/aescts/v2 v2.0.0
go: downloading gopkg.in/ini.v1 v1.66.3
go: downloading github.com/minio/md5-simd v1.1.2
go: downloading go.uber.org/multierr v1.7.0
go: downloading github.com/coreos/go-systemd/v22 v22.3.2
go: downloading github.com/jcmturner/rpc/v2 v2.0.3
go: downloading github.com/tklauser/go-sysconf v0.3.9
go: downloading github.com/davecgh/go-spew v1.1.1
go: downloading github.com/rcrowley/go-metrics v0.0.0-20201227073835-cf1acfcdf475
go: downloading github.com/eapache/go-xerial-snappy v0.0.0-20180814174437-776d5712da21
go: downloading github.com/eapache/go-resiliency v1.2.0
go: downloading github.com/eapache/queue v1.1.0
go: downloading github.com/jcmturner/goidentity/v6 v6.0.1
go: downloading github.com/tidwall/pretty v1.2.0
go: downloading github.com/tidwall/match v1.1.1
go: downloading github.com/go-openapi/strfmt v0.21.1
go: downloading github.com/go-openapi/jsonpointer v0.19.5
go: downloading github.com/josharian/intern v1.0.0
go: downloading github.com/tklauser/numcpus v0.3.0
go: downloading github.com/asaskevich/govalidator v0.0.0-20200907205600-7a23bdc65eef
go: downloading github.com/mitchellh/mapstructure v1.4.2
go: downloading github.com/go-openapi/errors v0.20.2
go: downloading go.mongodb.org/mongo-driver v1.7.5
go: downloading github.com/oklog/ulid v1.3.1
go: downloading github.com/matttproud/golang_protobuf_extensions v1.0.2-0.20181231171920-c182affec369
go: downloading github.com/go-openapi/jsonreference v0.19.6
go: downloading github.com/PuerkitoBio/purell v1.1.1
go: downloading github.com/lestrrat-go/option v1.0.0
go: downloading github.com/lestrrat-go/iter v1.0.1
go: downloading github.com/lestrrat-go/backoff/v2 v2.0.8
go: downloading github.com/lestrrat-go/blackmagic v1.0.0
go: downloading github.com/lestrrat-go/httpcc v1.0.0
go: downloading github.com/PuerkitoBio/urlesc v0.0.0-20170810143723-de5bf2ad4578
go: downloading cloud.google.com/go/iam v0.2.0
go: downloading github.com/googleapis/gax-go/v2 v2.1.1
go: downloading golang.org/x/xerrors v0.0.0-20200804184101-5ec99f83aff1
go: downloading go.opencensus.io v0.23.0
go: downloading github.com/golang/groupcache v0.0.0-20210331224755-41bb18bfe9da
go: downloading github.com/go-stack/stack v1.8.0
go: downloading github.com/go-openapi/runtime v0.21.1
go: downloading github.com/jessevdk/go-flags v1.4.0
go: downloading github.com/minio/mc v0.0.0-20220204044644-e048c85d71a7
go: downloading github.com/unrolled/secure v1.0.9
go: downloading github.com/go-openapi/validate v0.20.3
go: downloading github.com/docker/go-units v0.4.0
go: downloading github.com/charmbracelet/bubbletea v0.19.3
go: downloading github.com/charmbracelet/bubbles v0.10.0
go: downloading github.com/google/shlex v0.0.0-20191202100458-e7afc7fbc510
go: downloading github.com/charmbracelet/lipgloss v0.5.0
go: downloading maze.io/x/duration v0.0.0-20160924141736-faac084b6075
go: downloading github.com/pkg/profile v1.6.0
go: downloading github.com/posener/complete v1.2.3
go: downloading github.com/minio/colorjson v1.0.1
go: downloading github.com/pkg/xattr v0.4.4
go: downloading github.com/minio/filepath v1.0.0
go: downloading github.com/olekukonko/tablewriter v0.0.5
go: downloading gopkg.in/h2non/filetype.v1 v1.0.5
go: downloading golang.org/x/term v0.0.0-20210927222741-03fcf44c2211
go: downloading github.com/muesli/reflow v0.3.0
go: downloading github.com/lucasb-eyer/go-colorful v1.2.0
go: downloading github.com/muesli/termenv v0.11.1-0.20220204035834-5ac8409525e0
go: downloading github.com/containerd/console v1.0.2
go: downloading github.com/muesli/ansi v0.0.0-20211018074035-2e021307bc4b
go: downloading github.com/hashicorp/go-multierror v1.1.1
go: downloading github.com/hashicorp/errwrap v1.1.0
go: downloading cloud.google.com/go/compute v0.1.0
# github.com/minio/minio/cmd
cmd/iam.go:1031:40: undefined: expiredUser
cmd/iam.go:1049:39: undefined: expiredUser
make: *** [build] Error 2

```

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by donatello** [comment]
Added display name for use with user interfaces like console. This should be ready for review once again assuming tests pass.

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by harshavardhana** [comment]
please resolve the conflicts @donatello

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by kannappanr** [comment]
ping @harshavardhana @Alevsk @krisis

**Review by harshavardhana — APPROVED** [review]
The entire PR LGTM - however, I want to know how are we planning the following things

- to show the end-user what is the configured ARNs? to use?
- what is the potential plan for the Console UI to start using Role ARNs ?

**Comment by harshavardhana** [comment]
> The entire PR LGTM - however, I want to know how are we planning the following things
> 
> * to show the end-user what is the configured ARNs? to use?
> * what is the potential plan for the Console UI to start using Role ARNs ?

@donatello ^^ can you answer this?

**Comment by donatello** [comment]
> The entire PR LGTM - however, I want to know how are we planning the following things
> 
>     * to show the end-user what is the configured ARNs? to use?

Currently this is displayed in the console output after minio startup. I have started on a change to add higher level configuration support so we can query it via `mc admin identity` type commands.

>     * what is the potential plan for the Console UI to start using Role ARNs ?

For this, we can add some functions to query the IAM subsystem to return info needed to render the logins at UI (not present in this PR). It seems to be a better approach than creating more custom env vars for console UI to load.

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by harshavardhana** [comment]
@Alevsk @krisis can we get reviews on this?

**Comment by harshavardhana** [comment]
PTAL @vadmeste @krisis

**Comment by harshavardhana** [comment]
PTAL @Alevsk @dvaldivia

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

File: docs/sts/web-identity.md [hunk 4]
```diff
@@ -2,15 +2,90 @@
...
+```
+MINIO_IDENTITY_OPENID_DISPLAY_NAME="my first openid"
+MINIO_IDENTITY_OPENID_CONFIG_URL=http://myopenid.com/.well-known/openid-configuration
+MINIO_IDENTITY_OPENID_CLIENT_ID="minio-client-app"
+MINIO_IDENTITY_OPENID_CLIENT_SECRET="minio-client-app-secret"
+MINIO_IDENTITY_OPENID_SCOPES="openid,groups"
+MINIO_IDENTITY_OPENID_REDIRECT_URI="http://127.0.0.1:10000/oauth_callback"
+MINIO_IDENTITY_OPENID_ROLE_POLICY="consoleAdmin"
+
+MINIO_IDENTITY_OPENID_DISPLAY_NAME="another oidc"
```
Comment (krisis) [inline docs/sts/web-identity.md]: Shouldn't this env var name be `MINIO_IDENTITY_OPENID_DISPLAY_NAME_APP2`?

**Review by krisis — COMMENTED** [review]
LGTM :+1:  A minor typo in markdown docs.

**Comment by harshavardhana** [comment]
PTAL @donatello

File: docs/sts/web-identity.md (hunk 4 above)
Comment (donatello) [reply docs/sts/web-identity.md]: Will fix this.

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Review by krisis — APPROVED** [review]

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Pull request merged**